src/box/replication.h

```diff
@@ -402,6 +402,8 @@ struct replica {
 	bool is_relay_healthy;
 	/** Whether there is an applier subscribed to this replica. */
 	bool is_applier_healthy;
+	/** Whether the replica is connected right now. */
+	bool is_connected;
```

Comment:
Renamed it to `has_incoming_connection` and left a comment with elaboration about its difference from `is_relay_healthy`.

Reply:
Nope, that would make it inconsistent with `is_applier_healthy`. The flags are probably ok to have, but then it is confusing that we also directly check the relay status with `relay_state == FOLLOW`. Perhaps it would be a good idea to drop those checks completely, and introduce a new `enum replica_relay_state` and `enum replica_applier_state` which are stored and updated only in `replica` object, and replace all its flags. For example, when joining, we have `REPLICA_RELAY_CONNECTED`, and when subscribed - `REPLICA_RELAY_STREAMING`, etc.

Not necessary to do in this patch though.

Reply:
Another idea - why don't we drop `replica->relay` when it is unused? Right now replica always has a relay, whether it is started or not. But for applier it is different - we set it to `NULL` when the replica has no appliers.

I am wondering why not do the same with `replica->relay`. Make it `NULL` when no connection. And create it for each incoming connection. Then the flag could be replaced with `replica->relay != NULL`.

Reply:
> I am wondering why not do the same with `replica->relay`. Make it NULL when no connection.

Good idea! However, some minor questions appear. For example, when `replica->relay == NULL`, we don't know if it is equivalent of `RELAY_STOPPED` or `RELAY_OFF` status (it affects `box.info.replication` entry). Seems like relay rework is a perfect place to rework relay states as well, so I'd leave the `has_incoming_connection` flag for now. @locker, what do you think about it?

Reply:
Why do we have to distinguish between `RELAY_STOPPED` and `RELAY_OFF`? The exact state could perhaps be moved out of relay like I suggested above about `enum replica_relay_state`. So the `replica` and `relay` would be more detached from each other.

Yeah, I agree it is perhaps better to do it later during the relay rework.